### PR TITLE
BIG MERGE J-C-M branch to mainline

### PR DESCRIPTION
**Pull request by j-c-m**
This is the big merge discussed on gitter.

### CODE REVIEW

File: bot.user.js [hunk 1]
```diff
@@ -21,68 +19,73 @@ If not, see <http://www.gnu.org/licenses/>./gpl-howto.html
 // @grant        none
 // ==/UserScript==
 // Custom logging function - disabled by default
+
+window.scores = [];
```
Comment (ChadSki) [inline bot.user.js:23]: New global variable? Shouldn't this go in the userInterface module?

File: bot.user.js [hunk 2]
```diff
@@ -21,68 +19,73 @@ If not, see <http://www.gnu.org/licenses/>./gpl-howto.html
...
-    return w*w;
-};
-
-window.getX = function() {
-    return window.snake.xx;
-};
-window.getY = function() {
-    return window.snake.yy;
+window.getSnakeWidth = function(sc) {
+    if (sc === undefined) sc = window.snake.sc;
```
Comment (ChadSki) [inline bot.user.js:36]: I like this... pass a specific snake or default to our snake.

File: bot.user.js [hunk 3]
```diff
@@ -515,132 +668,34 @@ var bot = (function() {
...
-                            window.setAcceleration(1);
-                        }
-                    }
-                    window.kd_l = false;
-                    window.kd_r = false;
-                    canvas.setMouseCoordinates(window.goalCoordinates[0], window.goalCoordinates[1]);
+                    canvas.setMouseCoordinates(canvas.mapToMouse(window.goalCoordinates));
                 }
+            } else {
+                bot.tickCounter = -userInterface.framesPerSecond.getFPS();
```
Comment (ChadSki) [inline bot.user.js:698]: Heh. Cool trick to buy 1 second of time. Might add a comment for it?

File: bot.user.js [hunk 4]
```diff
@@ -861,43 +870,36 @@ var userInterface = (function() {
...
-            window.scroll_overlay.innerHTML = generalStyle + '(Mouse Wheel) Zoom in/out </span>';
-            window.quittomenu_overlay.innerHTML = generalStyle + '(Q) Quit to menu </span>';
-            window.changeskin_overlay.innerHTML = generalStyle + '(X) Change skin </span>';
-
-            window.clusteringchange_overlay.innerHTML = generalStyle + '(F) Switch clustering algorithm: ' + (bot.foodClusteringMode == 0 ? 'chancity' : 'ksofiyuk') + '</span>';
-            window.quickResp_overlay.innerHTML = generalStyle + '(ESC) Quick Respawn </span>';
+            window.collision_radius_multiplier_overlay.innerHTML = generalStyle + '(A/S) Collision radius multiplier: ' + window.collisionRadiusMultiplier + ' </span>';  
+        },
+
+        onFrameUpdate: function() {
```
Comment (ChadSki) [inline bot.user.js]: @ErmiyaEskandary is this the new eslint rule?

I think it wants us to say

    onFrameUpdate: function onFrameUpdate() {

File: bot.user.js [hunk 5]
```diff
@@ -122,18 +123,22 @@ var canvas = (function() {
             }
         },
 
-        // Draw a dot on the canvas.
-        drawDot: function(x, y, radius, colour, fill) {
+        // Draw a circle on the canvas.
+        drawCircle: function(circle, colour, fill, alpha) {
```
Comment (ChadSki) [inline bot.user.js:127]: :+1: much better name

**Comment by ermiyaeskandary** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/143/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

**Comment by ChadSki** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ErmiyaEskandary/Slither.io-bot/pull-request/143/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ErmiyaEskandary/Slither.io-bot)

File: bot.user.js (hunk 1 above)
Comment (j-c-m) [inline bot.user.js:23]: This was an in transition thing, just grabbing it from the console for now.

**Pull request merged**